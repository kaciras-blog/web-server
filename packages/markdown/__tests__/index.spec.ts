import { expect, it, vi } from "vitest";
import MarkdownIt from "markdown-it";
import MediaPlugin from "../lib/media.js";
import { Anchor, Classify, CollectLinks, Footnote } from "../lib/index.js";

it("should add class to inlined code block", () => {
	const markdownIt = new MarkdownIt();
	markdownIt.use(Classify);

	expect(markdownIt.render("`foobar`")).toMatchSnapshot();
});

it("should add anchor to titles", () => {
	const markdownIt = new MarkdownIt();
	markdownIt.use(Anchor);

	expect(markdownIt.render("# foobar")).toMatchSnapshot();
});

it("should support footnote syntax", () => {
	const markdownIt = new MarkdownIt();
	markdownIt.use(Footnote);

	const html = markdownIt.render("test[^1]\n\n[^1]: foobar\n\nafter");
	expect(html).toMatchSnapshot();
});

it("should collect file links", () => {
	const handler = vi.fn();
	const markdownIt = new MarkdownIt();
	markdownIt.use(MediaPlugin);
	markdownIt.use(CollectLinks, handler);

	const lines = [
		"[](//example.com:123/some-file)",
		"![](/foo?a=b)",
		"![empty]()",
		"@video[](/bar)",
		"@gif[](http://example.com/favicon.ico)",

		"# H1\nfoobar",
	];
	markdownIt.render(lines.join("\n\n"));

	expect(handler).toHaveBeenCalledTimes(5);
	expect(handler).toHaveBeenCalledWith("//example.com:123/some-file");
	expect(handler).toHaveBeenCalledWith("/foo?a=b");
	expect(handler).toHaveBeenCalledWith("");
	expect(handler).toHaveBeenCalledWith("/bar");
	expect(handler).toHaveBeenCalledWith("http://example.com/favicon.ico");
});
