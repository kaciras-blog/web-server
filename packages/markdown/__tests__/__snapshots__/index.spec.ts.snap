// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should add anchor to titles 1`] = `
"<h1 id=\\"foobar\\" tabindex=\\"-1\\">foobar <a class=\\"anchor-link\\" href=\\"#foobar\\">#</a></h1>
"
`;

exports[`should add class to inlined code block 1`] = `
"<p><code class=\\"inline-code\\">foobar</code></p>
"
`;

exports[`should support footnote syntax 1`] = `
"<p>test<sup class=\\"footnote-ref\\"><a href=\\"#fn1\\" id=\\"fnref1\\">[1]</a></sup></p>
<p>after</p>
<h2 class='footnotes'>参考</h2><ol class='footnotes-list'><li id=\\"fn1\\" class=\\"footnote-item\\"><p>foobar <a href=\\"#fnref1\\" class=\\"footnote-backref\\">↩︎</a></p>
</li>
</ol>"
`;
